src/main/kotlin/com/openlattice/chronicle/jobs/BackgroundChronicleDeletionService.kt

```diff
@@ -0,0 +1,73 @@
+package com.openlattice.chronicle.jobs
+
+import com.openlattice.chronicle.auditing.AuditedOperationBuilder
+import com.openlattice.chronicle.auditing.AuditingManager
+import com.openlattice.chronicle.postgres.ResultSetAdapters
+import com.openlattice.chronicle.services.jobs.JobService
+import com.openlattice.chronicle.storage.ChroniclePostgresTables.Companion.JOBS
+import com.openlattice.chronicle.storage.PostgresColumns.Companion.CREATED_AT
+import com.openlattice.chronicle.storage.PostgresColumns.Companion.JOB_DATA
+import com.openlattice.chronicle.storage.PostgresColumns.Companion.JOB_ID
+import com.openlattice.chronicle.storage.StorageResolver
+import org.slf4j.LoggerFactory
+import org.springframework.scheduling.annotation.Scheduled
+import java.sql.Connection
+
+/**
+ * @author Solomon Tang <solomon@openlattice.com>
+ */
+class BackgroundChronicleDeletionService(
+  private val jobService: JobService,
+  private val storageResolver: StorageResolver,
+  private val auditingManager: AuditingManager,
+) {
+    companion object {
+        private val logger = LoggerFactory.getLogger(BackgroundChronicleDeletionService::class.java)!!
+
+        private val GET_NEXT_JOB_SQL = """
+            DELETE FROM ${JOBS.name}
```

Comment:
Rather than deleting you should mark a job as done and have a separate background job to clear out done jobs after an expiration period ( if any ). Also need a way to track deletion jobs on some sort of basis. i.e  a user should be able to say give me a list of all deletion jobs in the last two weeks and report on successful execution and number of items deleted.